marklogic-rdf4j/src/main/java/com/marklogic/semantics/rdf4j/client/MarkLogicClientImpl.java

```diff
@@ -596,4 +678,333 @@ private static SPARQLQueryDefinition bindObject(SPARQLQueryDefinition qdef, Stri
+                }
+            }
+
+            @Override
+            public void handleComment(String comment) throws RDFHandlerException {
+
+            }
+
+            private void triple(StringBuilder sb, Statement st) {
+                sb.append("<sem:triple>\n");
+                sb.append(subject(st.getSubject()));
+                sb.append(predicate(st.getPredicate()));
+                sb.append(object(st.getObject()));
+                sb.append("</sem:triple>\n");
+            }
+        });
+    }
+
+    private String subject(Value subject) {
+        if (subject instanceof org.eclipse.rdf4j.model.BNode) {
+            //Skolemization using toString()
+            return "<sem:subject>http://marklogic.com/semantics/blank/" + subject.toString() + "</sem:subject>\n";
+        } else {
+            return "<sem:subject>" + subject.stringValue() + "</sem:subject>\n";
+        }
+    }
+
+    private String predicate(Value predicate) {
+        return "<sem:predicate>" + predicate.stringValue() + "</sem:predicate>\n";
+    }
+
+    private String object(Value object) {
+        if (object instanceof Literal) {
+            Literal lit = (Literal) object;
+            String lang = lit.getLanguage().orElse("");
+            String type = lit.getDatatype().toString();
+
+            if (!lang.equals("")) {
+                lang = " xml:lang=\"" + lang + "\"";
+            }
+
+            if ("".equals(lang)) {
+                if (type == null) {
+                    type = "http://www.w3.org/2001/XMLSchema#string";
+                }
+                type = " datatype=\"" + type + "\"";
+            } else {
+                type = "";
+            }
+
+            return "<sem:object" + type + lang + ">" + object.stringValue() + "</sem:object>\n";
+        } else if (object instanceof org.eclipse.rdf4j.model.BNode) {
+            //Skolemization using toString()
+            return "<sem:object>http://marklogic.com/semantics/blank/" + object.toString() + "</sem:object>\n";
+        } else {
+            return "<sem:object>" + object.stringValue() + "</sem:object>\n";
+        }
+    }
+
+    private String escapeXml(String _in) {
+        if (null == _in) {
+            return "";
+        }
+        return patterns[2].matcher(
+                patterns[1].matcher(
+                        patterns[0].matcher(_in).replaceAll("&amp;"))
+                        .replaceAll("&lt;")).replaceAll("&gt;");
+    }
+
+    private void insertGraphDocuments(Transaction tx, ThreadPoolExecutor executor, List<Future<?>> futures, Set<String> graphSet){
+        int MAX_GRAPHS_PER_REQUEST = 100;
+        int max = MAX_GRAPHS_PER_REQUEST;
+        StringBuilder stringBuilder = new StringBuilder();
+
+        for(String graph: graphSet){
+            if(max == 1){
+                max = MAX_GRAPHS_PER_REQUEST;
+                stringBuilder.append("CREATE SILENT GRAPH <").append(escapeXml(graph)).append(">;");
```

Comment:
So it's good that you coded this just to see how it might be done but you actually did the wrong encoding.  As an IRI/URI, these should be just encoded with 

```
URLEncoder.encode(
    "urlParameterString",
    java.nio.charset.StandardCharsets.UTF_8.toString()
  )
```

So I'll request you take out this private method and replace with static calls to java.net.URLEncoder.  it's part of standard library